services/motion/builtin/builtin.go

```diff
@@ -237,7 +270,118 @@ func (ms *builtIn) MoveOnGlobe(
 ) (bool, error) {
-	return false, ErrNotImplemented
+	operation.CancelOtherWithLabel(ctx, motionOpId)
+
+	// create a new empty framesystem which we will add our base to
+	fs := referenceframe.NewEmptyFrameSystem("")
+
+	// build maps of input enabled resources
+	_, resources, err := ms.fsService.CurrentInputs(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	// get the localizer
+	local, ok := ms.localizers[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	// assert localizer as a movementSensor to get map limits
+	movementSensor, ok := local.(movementsensor.MovementSensor)
+	if !ok {
+		return false, fmt.Errorf("cannot assert local of type %T as a movementSensor", local)
+	}
+
+	currentPosition, _, err := movementSensor.Position(ctx, nil)
+	if err != nil {
+		return false, err
+	}
+	currentPose := spatialmath.GeoPointToPose(currentPosition)
+
+	// convert destination into a spatialmath pose in frame with respect to 0 latitude, 0 longitude
+	dstPose := spatialmath.GeoPointToPose(destination)
+	dstPIF := referenceframe.NewPoseInFrame(referenceframe.World, dstPose)
+
+	// convert GeoObstacles into GeometriesInFrame then into a Worldstate
+	geoms := spatialmath.GeoObstaclesToGeometries(obstacles)
+	gif := referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)
+	wrldst, err := referenceframe.NewWorldState([]*referenceframe.GeometriesInFrame{gif}, nil)
+	if err != nil {
+		return false, err
+	}
+	// get worldstate limits
+	limits := wrldst.BoundingBox(ctx)
+
+	// recalculate limits with destination and current position
+	minX := math.Min(math.Min(currentPose.Point().X, dstPose.Point().X), limits[0].Min)
+	maxX := math.Min(math.Max(currentPose.Point().X, dstPose.Point().X), limits[0].Max)
+	minY := math.Min(math.Min(currentPose.Point().Y, dstPose.Point().Y), limits[1].Min)
+	maxY := math.Min(math.Max(currentPose.Point().Y, dstPose.Point().Y), limits[1].Max)
+	limits = []referenceframe.Limit{
+		{Min: minX, Max: maxX},
+		{Min: minY, Max: maxY},
+		{Min: -2 * math.Pi, Max: 2 * math.Pi},
+	}
+
+	// create a KinematicBase from the componentName
+	baseComponent, ok := ms.components[componentName]
+	if !ok {
+		return false, fmt.Errorf("only Base components are supported for MoveOnMap: could not find an Base named %v", componentName)
+	}
+	kw, ok := baseComponent.(base.KinematicWrappable)
+	if !ok {
+		return false, fmt.Errorf("cannot move base of type %T because it is not KinematicWrappable", baseComponent)
+	}
+	kb, err := kw.WrapWithKinematics(ctx, local, limits)
+	if err != nil {
+		return false, err
+	}
+
+	// check for cancelled context before we are start planning
+	if ctx.Err() != nil {
+		ms.logger.Info("successfully canceled motion service MoveOnGlobe")
+		return true, ctx.Err()
+	}
+
+	// get current position
+	inputs, err := kb.CurrentInputs(ctx)
+	if err != nil {
+		return false, err
+	}
+	seedMap := map[string][]referenceframe.Input{kb.ModelFrame().Name(): inputs}
```

Comment:
We've named things seedMap in the past but its really not a descriptive name so I'd like to move away from it.  how about `inputMap` `fsInputs` or something like that?

Reply:
i like the sound of `inputMap`